### DISCUSSION TITLE
Sending data from webview to sidecar

### DISCUSSION THREAD

**Discussion by tomg7** [tauri-apps/tauri#4664] [Q&A]
I have the sidecar example running locally and I see how the webview can get data from the sidecar both by spawning the sidecar from the webview (index.html) and from the rust backend (main.rs).
I can also see how the rust backend sends a message to the sidecar using `child.write` in response to a Stdout sidecar event, but I cannot figure out how to send a message to the sidecar from the webview when the user clicks a button. 

I tried to send a message directly from the webview:
`const child = await command.spawn();`
`await child.write('Message from webview');`
I see the child object and its write function in DevTools and I get no errors, but I do not see the example nodejs sidecar app logging the message.
 
I also tried to call `child.write` from a `tauri::command` or from a backend event listener, but I am new to rust and I could not figure out how to refer `child`, that is created inside `tauri::async_runtime::spawn`, from a command or an event listener without getting errors about borrowing.

**Comment by Xoffio** [comment]
@tomg7 [This](https://docs.rs/tauri/1.0.3/tauri/api/process/struct.Command.html#method.spawn) might help

**Reply by tomg7** [reply]
@SpaceComet Thanks for the link but it is the same code as you can find in the sidecar example. What I need is the `child.write` command working in an Event Listener or in a Tauri Command outside of the `tauri::async_runtime::spawn` block:

1. Event Listener
```
let id = window.listen("event", |event| {
  child.write("message from Rust\n".as_bytes()).unwrap();
});
````

2. Tauri Command
```
#[tauri::command]
fn send_msg(msg: &str) {
  child.write(msg.as_bytes()).unwrap();
});
````

My issue is that I cannot reference `child` (that lives in the `tauri::async_runtime::spawn` block and refers to the spawned child process) from the Event Listener or from the Tauri Command. 

It is not a Tauri issue I just do not know how to do this in Rust. I have not worked with Rust code before.
I tried to declare child outside of the `tauri::async_runtime::spawn` block or assign child to a variable declared in `main()`  but I get all sorts of errors about borrowing and moving ownership. I also looked at smart pointers/context/state/mutex examples but I do not know how to use those here. 

I would like to port an Electron app to Tauri and all I need is bidirectional communication between the front-end and the Nodejs sidecar, but I could not find an example showing how to send an event, for example when the user clicks a button, from the front-end to the sidecar process.

**Reply by Xoffio** [reply]
@tomg7 I see... It's quite hard to help you with out some more code... 

So you already spawn the command in the main? can't you just spawn the program inside the function?
